core/shared/src/main/scala/scalaz/zio/RefM.scala

```diff
@@ -34,7 +34,7 @@ import scalaz.zio.Exit.Cause
  * } yield ()
  * }}}
  */
-final class RefM[A] private (value: Ref[A], queue: Queue[RefM.Bundle[A, _]]) extends Serializable {
+final class RefM[R, A] private (value: Ref[A], queue: Queue[RefM.Bundle[R, A, _]]) extends Serializable {
```

Comment:
Let's not add it to the class `RefM`, but rather, the methods themselves. This will provide more polymorphism.

Reply:
I don't think this is possible with the current implementation of `RefM`: the methods are just adding the io to a queue, but the queue creation and the `run` method (the one that actually runs them) are called from the `make`.